src/LibraryManager.Vsix/Json/Completion/LibraryIdCompletionProvider.cs

```diff
@@ -71,17 +71,15 @@ protected override IEnumerable<JSONCompletionEntry> GetEntries(JSONCompletionCon
 
             if (task.IsCompleted)
             {
-                CompletionSet set = task.Result;
-                int start = member.Value.Start;
-                ITrackingSpan trackingSpan = context.Snapshot.CreateTrackingSpan(start + 1 + set.Start, set.Length, SpanTrackingMode.EdgeInclusive);
+                CompletionSet completionSet = task.Result;
 
-                if (set.Completions != null)
+                if (completionSet.Completions != null)
                 {
-                    foreach (CompletionItem item in set.Completions)
+                    List<JSONCompletionEntry> results = GetCompletionList(member, context, completionSet, count);
+
+                    foreach (JSONCompletionEntry completionEntry in results)
                     {
-                        string insertionText = item.InsertionText.Replace("\\\\", "\\").Replace("\\", "\\\\");
-                        ImageMoniker moniker = item.DisplayText.EndsWith("/") || item.DisplayText.EndsWith("\\") ? _folderIcon : _libraryIcon;
-                        yield return new SimpleCompletionEntry(item.DisplayText, insertionText, item.Description, moniker, trackingSpan, context.Session, ++count);
+                        yield return (completionSet.CompletionType == CompletionSortOrder.Version) ? completionEntry as VersionCompletionEntry : completionEntry as SimpleCompletionEntry;
```

Comment:
>(completionSet.CompletionType == CompletionSortOrder.Version) ? completionEntry as VersionCompletionEntry : completionEntry as SimpleCompletionEntry; [](start = 37, length = 149)

Why do you need this cast?

Reply:
I see. I don't need to deal with the completionEntry type as it's already initialized.

---
In reply to: [205855293](https://github.com/aspnet/LibraryManager/pull/312#discussion_r205855293) [](ancestors = 205855293)